src/components/FileDropZone.tsx

```diff
@@ -14,54 +23,119 @@ function convertToPlainText(html: string): string {
 }
 /* eslint-enable functional/immutable-data */
 
-function createSourceFromParagraphs(
-  paragraphs: Array<string>,
+function createRelationsFromParagraphNodes(
+  nodes: KnowNode[],
   myself: PublicKey
-): {
-  nodes: KnowNode[];
-  relations: Relations;
-  topNodeID: LongID;
-} {
-  const topParagraph = paragraphs[0];
-  const furtherParagraphs = paragraphs.slice(1);
+): [relations: Relations, topNodeID: LongID] {
+  const topParagraph = nodes[0];
+  const furtherParagraphs = nodes.slice(1);
+  const relations = bulkAddRelations(
+    newRelations(topParagraph.id, "", myself),
```

Comment:
i don't get why the type of the head relations is `""`. is this a general pattern?

Reply:
"" is the default relation